R/plotPValHist.R

```diff
@@ -25,111 +24,150 @@
 #' }
 #'
 #' @import ggplot2
-#' @importFrom dplyr filter
+#' @importFrom dplyr filter select
 #' @importFrom canvasXpress canvasXpress
 #'
 #' @export
 plotPvalHist <- function(P.Val,
                          plotType = "canvasXpress",
-                         facet = TRUE,
-                         binWidth = 0.02,
-                         alpha = 0.6,
-                         color = "dodgerblue3",
-                         fill = "dodgerblue3") {
+                         facet          = TRUE,
+                         binWidth       = 0.02,
+                         transparency   = 0.6,
+                         color    = "dodgerblue3") {
 
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
+    plotType = tolower(plotType)
+
+    assertthat::assert_that(!missing(P.Val),
+                            class(P.Val)[[1]] %in% c("matrix","data.frame"),
+                            msg = "P.Val must be specified and must be of class matrix or dataframe.")
+
+    assertthat::assert_that(all(sapply(data.frame(P.Val),is.numeric)),
```

Comment:
both discreate and continuous values are numeric,  so how you are going to distinct between them?

Reply:
You cannot bin a character variable. As long as the value is numeric , the plot works fine.